### PR TITLE
Simplify spheremap shaders

### PR DESCRIPTION
**Pull request by bcamper**
This simplifies the spheremap shader to avoid the eye / reflections calculations, which it turns out can be avoided because we are using a simplified concept of environment map where the "eye" is always looking straight down.

It also reduces shader duplication by using `mix` and moving a parameter into a uniform (`u_zoom_scale`). Because Tangram now shares source-identical program objects, this will avoid an additional program compile.

### CODE REVIEW

**Comment by bcamper** [comment]
cc @matteblair @tallytalwar let's try this on Tizen too

**Comment by bcamper** [comment]
Differ test looks good (no change):

http://tangrams.github.io/differ/?1=https://github.com/tangrams/walkabout-style/blob/gh-pages/walkabout-style.yaml&2=https://github.com/tangrams/walkabout-style/blob/simplify-spheremap/walkabout-style.yaml&lib1=0.12&lib2=&go

File: walkabout-style.yaml [hunk 1]
```diff
@@ -913,75 +913,41 @@ styles:
...
-                    //float scale = 10./(u_map_position.z-.7) + 1.8;
-                    float scale1 = 20./(u_map_position.z) + 1.5;
-                    normal.z /= scale1; // turn terrain exaggeration up/down
+                    // turn terrain exaggeration up/down
                     // fade out spheremap normals with a function
-                    // https://www.desmos.com/calculator/ptgkzcnfyc
-                    float m = 3.5 * (u_map_position.z - 0.8) * pow(e, u_map_position.z * -.29);
+                    const float e = 2.71828;
+                    float scale1 = 20./(u_map_position.z) + 1.5;
+                    float m = u_zoom_scale * (u_map_position.z - 0.8) * pow(e, u_map_position.z * -.29);
```
Comment (matteblair) [inline walkabout-style.yaml]: Instead of `pow(base, exponent)` with a constant base `e`, we can use `exp(exponent)`.

Comment (bcamper) [reply walkabout-style.yaml]: 👍

**Comment by nvkelso** [comment]
LGTM after verifying browsing the map.

**Pull request merged**